controllers/reconcilers/configuration/prometheus.go

```diff
@@ -455,7 +456,7 @@ func (r *Reconciler) reconcilePrometheus(ctx context.Context, cr *v1.Observabili
 			Resources:                       model.GetPrometheusResourceRequirement(cr),
 		}
 		if cr.Spec.Storage != nil && cr.Spec.Storage.PrometheusStorageSpec != nil {
-			prometheus.Spec.Storage = cr.Spec.Storage.PrometheusStorageSpec
+			prometheus.Spec.Storage, err = getPrometheusStorageSpecHelper(cr, indexes)
```

Comment:
`err` is not checked here. Better to do call the function outside of the assignment.